sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -1405,6 +1409,66 @@ public Flux<ByteBuffer> toFluxByteBuffer() {
         return content.toFluxByteBuffer();
     }
 
+    /**
+     * Writes the contents of this {@link BinaryData} to the given {@link OutputStream}.
+     * <p>
+     * This method does not close the {@link OutputStream}.
+     * <p>
+     * The contents of this {@link BinaryData} will be written without buffering. If the underlying data source isn't
+     * {@link #isReplayable()}, after this method is called the {@link BinaryData} will be consumed and can't be read
+     * again. If it needs to be read again, use {@link #toReplayableBinaryData()} to create a replayable copy.
+     *
+     * @param outputStream The {@link OutputStream} to write the contents of this {@link BinaryData} to.
+     * @throws NullPointerException If {@code outputStream} is null.
+     * @throws IOException If an I/O error occurs.
+     */
+    public void writeTo(OutputStream outputStream) throws IOException {
+        Objects.requireNonNull(outputStream, "'outputStream' cannot be null.");
+
+        content.writeTo(outputStream);
```

Comment:
I really like `writeTo` APIs!
It seems we don't have much test coverage for them on binaryData/Content, mind adding some tests?

Reply:
Moved most changes into shared code paths with existing tests